### PR TITLE
Adding BLE protocol engine configuration to maestro conf

### PR DESCRIPTION
**Pull request by ygoyal18**
Created new process template in maestro conf for BLE protocol engine
    
    1. Adding BLE protocol engine configuration to maestro conf
    2. To avoid "Allocation failed out of memory" error, created new
       node process template and allowing this process to increase upto
       1GB of memory.

### CODE REVIEW

**Comment by theotherjimmy** [comment]
Could you split f4fabe7 into 2 commits one for each of:
 * add configuration for EP gateway
 * increase the heap size for all node processes

?

----------------------

If you like, I can do that for you.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    params:
+      identityPath: "/userdata/edge_gw_config/identity.json"
+# relaymq:
+#     host: "{{ARCH_relaymqHost}}"
+#     logging: true
+#     client_cert: "{{ARCH_CLIENT_CERT_PEM}}"
+#     client_key: "{{ARCH_CLIENT_KEY_PEM}}"
+#     ---Only specify server_name if the RootCA is not in the default list
+#     This is only a valid option if you are not using a well-known CA
+#    server_name: "devcloud.wigwag.io"
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: Let's not add commented out code into a new file.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+   - key: "USER_VERSIONS_FILE"
+     value: "/mnt/.overlay/user/slash/wigwag/etc/versions.json"
+   - key: "UPGRADE_VERSIONS_FILE"
+     value: "/mnt/.overlay/upgrade/wigwag/etc/versions.json"
+mdns:
+  # disable: true
+  static_records:
+   - name: "WigWagRelay"
+     service: "_wwservices._tcp"  # normally something like https or ftp
+     # domain: "local"     # local is default
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: This is not needed. Could it be removed.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+     value: "/mnt/.overlay/factory/wigwag/etc/versions.json"
+   - key: "USER_VERSIONS_FILE"
+     value: "/mnt/.overlay/user/slash/wigwag/etc/versions.json"
+   - key: "UPGRADE_VERSIONS_FILE"
+     value: "/mnt/.overlay/upgrade/wigwag/etc/versions.json"
+mdns:
+  # disable: true
+  static_records:
+   - name: "WigWagRelay"
+     service: "_wwservices._tcp"  # normally something like https or ftp
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: Could you remove this comment, it's not useful.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+   - key: "RELAY_VERSIONS_FILE"
+     value: "/wigwag/etc/versions.json"
+   - key: "FACTORY_VERSIONS_FILE"
+     value: "/mnt/.overlay/factory/wigwag/etc/versions.json"
+   - key: "USER_VERSIONS_FILE"
+     value: "/mnt/.overlay/user/slash/wigwag/etc/versions.json"
+   - key: "UPGRADE_VERSIONS_FILE"
+     value: "/mnt/.overlay/upgrade/wigwag/etc/versions.json"
+mdns:
+  # disable: true
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: Let's just skip this line too.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+     format_level: "\"level\":\"%s\", "
+     format_tag: "\"tag\":\"%s\", "
+     format_origin: "\"origin\":\"%s\", "
+     format_pre_msg: "\"text\":\""
+     format_post: "\"},"
+     flag_json_escape_strings: true
+     filters:
+       - levels: all
+         format_pre: "{"     # you will wrap this output with { "log": [ OUTPUT ] }
+#   - file: "{{TMP_DIR}}/maestro.log"
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: Remove this, and the next 52 lines, too please

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 6]
```diff
@@ -0,0 +1,426 @@
...
+     output_file: "{{SSL_CERTS_PATH}}/server.cert.pem"
+   - name: "ca_chain"
+     template: "{{ARCH_CA_CHAIN_CERT_PEM}}"
+     output_file: "{{SSL_CERTS_PATH}}/ca-chain.cert.pem"
+container_templates:
+   - name: "deviceJS_process"
+     immutable: true  # don't store in DB
+     depends_on:
+        - "devicejs"
+     cgroup:                 # will implement later
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: This configuration actually does nothing. Please remove it.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 7]
```diff
@@ -0,0 +1,426 @@
...
+        mem_limit: 10000000
+     die_on_parent_death: true
+     inherit_env: true
+     add_env:
+        - "LD_PRELOAD=/usr/lib/libcrypto.so.1.0.2"
+        - "DEVJS_ROOT={{DEVICEJS_ROOT}}"
+        - "DEVJS_CONFIG_FILE=/wigwag/etc/devicejs/devicejs.conf"
+        - "NODE_PATH={{WIGWAG_NODE_PATH}}"
+     exec_cmd: "{{NODE_EXEC}}"        # will use PATH if not absolute path (as per execvp())
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: Please remove this superfluous comment.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 8]
```diff
@@ -0,0 +1,426 @@
...
+     die_on_parent_death: true
+     exec_cmd: "/wigwag/system/bin/devicedb"
+     restart: true
+     restart_limit: 5000
+     restart_pause: 1
+     exec_args:
+        - "start"
+        - "-conf=/wigwag/etc/devicejs/devicedb.yaml"
+   - job: "devicejs"
+     defer_running_status: 30000   # waits 8 seconds before marking this Job as RUNNING, which delays all dependencies
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: Is this comment correct? it looks like it waits 30 seconds.

File: conf/maestro-conf/ep-developer-edge-config.yaml [hunk 9]
```diff
@@ -0,0 +1,426 @@
...
+     restart: true
+     restart_limit: 5000
+     restart_pause: 90
+     config: |
+        {
+            "debug": true,
+            "mapUnplacedDevices": true,
+            "ignoreDevicesById" : [],
+            "relayID": "{{ARCH_SERIAL_NUMBER}}",
+            "mbedAPIKey": "ak_1MDE1ZDUwYTFmYmM2MDI0MjBhMDExMTA5MDAwMDAwMDA015e967fe74802420a01390300000000aJOhYRqzTXINzN9Aw2OkV5WrG0YwgmKS",
```
Comment (theotherjimmy) [inline conf/maestro-conf/ep-developer-edge-config.yaml]: This looks like it's not something we should commit.

**Comment by ygoyal18** [comment]
@theotherjimmy I have made the changes to the PR. We can resolve the above comments are we have removed the file in question.

**Comment by theotherjimmy** [comment]
Waiting on doing a `meta-pelion-edge` build with these changes.

**Pull request merged**